### PR TITLE
Fixes #2121 - Allow players to teleport into a location with water if configured

### LINKED ISSUE #2121: (1.13) Teleporting to a home/spawn that is in water from another world puts you in the wrong place.
Within the same world, you get put in the water at the right coordinates. 

From another world, you get put in the nearest air block above the X, Z coordinate.

Re-sending the command will put you in the proper place.

### PR DESCRIPTION
**Pull request by caojohnny**
This is a continuation of #2457, accidentally hit rebase and recommitted a load of commits from the 2.x log into my repo...

I've reverted the code and added the configuration option, modifying the `LocationUtil#HOLLOW_MATERIALS` as necessary when the config is loaded.

New demo: https://streamable.com/pm50r
```
[16:01:00 INFO]: Server version: 1.13.2-R0.1-SNAPSHOT git-Spigot-3cb9dcb-77ca7ca (MC: 1.13.2)
[16:01:00 INFO]: EssentialsX version: 2.16.1.154
[16:01:00 INFO]: Vault is not installed. Chat and permissions may not work.
```

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/Settings.java [hunk 1]
```diff
@@ -1540,4 +1550,18 @@ private boolean _allowOldIdSigns() {
     public boolean allowOldIdSigns() {
         return allowOldIdSigns;
     }
+
+    private boolean isWaterSafe;
+
+    private boolean _isWaterSafe() {
+        boolean isWaterSafe = config.getBoolean("is-water-safe", false);
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Settings.java]: I'd rename this to `_isWaterSafe` to aid readability.

Comment (caojohnny) [reply Essentials/src/com/earth2me/essentials/Settings.java]: What are you proposing to rename? The method or the field or the config key?

Comment (mdcfe) [reply Essentials/src/com/earth2me/essentials/Settings.java]: The variable inside the method. It has the same name as the field above but it isn't clear at first glance.

File: Essentials/src/com/earth2me/essentials/utils/LocationUtil.java [hunk 2]
```diff
@@ -16,11 +16,18 @@
 
 
 public class LocationUtil {
+    // Water types used for TRANSPARENT_MATERIALS and is-water-safe config option
+    private static final EnumSet<Material> WATER_TYPES = EnumSet.of(Material.WATER);
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/utils/LocationUtil.java]: Is there a reason for not using `EnumUtil.getAllMatching`?

**Review by mdcfe — DISMISSED** [review]

**Review by drtshock — DISMISSED** [review]

**Review by mdcfe — APPROVED** [review]

**Pull request merged**